packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -615,6 +615,12 @@ export abstract class BaseService extends Resource
         && deploymentController.type !== DeploymentControllerType.ECS) {
       Annotations.of(this).addError('Deployment circuit breaker requires the ECS deployment controller.');
     }
+
+    if (props.deploymentAlarms
```

Comment:
Ok 
so to validate the final state of the resource we can directly throw an error in the validation for deployment alarms,
As Annotations used in the circuit breaker I thought we need to use that in the deployment alarms,
Now I got the point 
So instead of using  `Annotations.of(this).addError` we can use `throw new Error ('Deployment circuit breaker requires the ECS deployment controller.');`

thanks for the clarification.